pkg/testing/kubernetes/kind/provisioner.go

```diff
@@ -72,46 +76,33 @@ func (p *provisioner) SetLogger(l runner.Logger) {
 }
 
 func (p *provisioner) Supported(batch define.OS) bool {
-
-	supported := batch.Type == define.Kubernetes && batch.Arch == runtime.GOARCH && (batch.Distro == "" || batch.Distro == "kind")
-
-	if supported && batch.Version != "" {
```

Comment:
the Version logical check here was to allow a k8s test to run only for a specific k8s version. I admit that being able to define version thresholds and not fixed one would be more handy, but with that said, do we want remove this capability?

Reply:
I didn't understand why it was there. That makes more sense. I will add it back, with a code comment so the next developer doesn't remove it.

Reply:
I take that back, my change auto creates a kind cluster for that version of k8s even in the environment variable has a `K8S_VERSION` that is different. `K8S_VERSION` in this case is only the default in the case that a test is not specific.

Reply:
ok based on this comment and https://github.com/elastic/elastic-agent/pull/5547#discussion_r1765159296 I just want to understand which layer holds responsibility for these testing knobs, for example; the "supported" k8s version(s) is mandated by the env vars and if a test doesn't support them it is just skipped [CI pipeline responsibility] or based on the defined versions of the k8s tests these are the versions of the clusters that are gonna be defined [developer writing the test responsibility]?! The same goes for k8s providers and docker images. Personally I more accustomed to defining what is allowed or not through env vars [CI pipeline responsibility] and tests either getting skipped or running based on the former. But here it is seems to me that we are mix matching them. Maybe I am missing something!? 🙂

Reply:
The more I think about this, the kubernetes mode of the integration testing suite is very different then how the standard mode works. The standard modes defines everything in code, and then a caller can say what they want to allow to run. That is defined by the testing runner even before the provider is invoked.

That allows for things like `mage integration:matrix` to perform running a larger set of tests across a whole matrix of versions. It think I need to adjust this PR to take that same approach and `mage integration:matrix` would then run multiple versions of kubernetes, but the standard `mage integration:test` would only run a single version.

That would make this align correctly with the standard operating mode of the integration suite.

Reply:
ok from this comment I understand that the knobs are controlled by the defines in each test and, if that's the case indeed, then yes the k8s integration tests need some shifting towards that direction. You wanna leave that for a subsequent PR or tackle it in this one?